src/tree/assess/TreeAssessor.js

```diff
@@ -57,19 +59,49 @@ class TreeAssessor {
 	 * @param {Paper}                      paper The paper to assess. This contains metadata about the text.
 	 * @param {module:tree/structure.Node} node  The text to check.
 	 *
-	 * @returns {Promise<number>} The overall assessment result.
+	 * @returns {Promise<{results: AssessmentResult[], score: number}>} The assessment results and the overall score.
 	 */
 	async assess( paper, node ) {
 		const applicableAssessments = await this.getApplicableAssessments( paper, node );
 		/*
 		  Apply every applicable assessment on the document.
 		  Wait before they are done before aggregating the results
-		  and returning the final score.
+		  and returning the results and final score.
 		 */
 		const results = await Promise.all(
-			applicableAssessments.map( assessment => assessment.apply( paper, node ) )
+			applicableAssessments.map( assessment => this.applyAssessment( assessment, paper, node ) )
+		);
+		// Filter out errored assessments.
+		const validResults = results.filter( result => result.getScore() !== -1 );
+		// Compute overall score.
+		const overallScore = this.scoreAggregator.aggregate( validResults );
+
+		return {
+			results: results,
+			score: overallScore,
+		};
+	}
+
+	/**
+	 * Applies the given assessment to the paper-node combination.
+	 *
+	 * @param {module:tree/assess.Assessment} assessment The assessment to apply.
+	 * @param {Paper}                         paper      The paper to apply the assessment to.
+	 * @param {module:tree/structure.Node}    node       The root node of the tree to apply the assessment to.
+	 *
+	 * @returns {Promise<AssessmentResult>} The result of the assessment.
+	 */
+	async applyAssessment( assessment, paper, node ) {
+		return assessment.apply( paper, node ).catch(
+			() => {
+				return new AssessmentResult( {
+					text: this.i18n.sprintf(
+						/* Translators: %1$s expands to the name of the assessment. */
+						this.i18n.dgettext( "js-text-analysis", "An error occurred in the '%1$s' assessment" ) ),
```

Comment:
You should probably also add the assessment's name to the `sprintf` call then 😉

Reply:
Good catch 😅 .